manticore/platforms/linux.py

```diff
@@ -3204,7 +3201,7 @@ def sys_accept(self, sockfd, addr, addrlen):
         # TODO: Make a concrete connection actually an option
         # return super().sys_accept(sockfd, addr, addrlen)
 
-    def sys_open(self, buf: int, flags: int, mode: int) -> int:
+    def sys_open(self, buf: int, flags: int, mode: Optional[int]) -> int:
```

Comment:
We don't appear to actually use `mode` for anything. It gets passed around a few times but ultimately ignored in favor of `flags`. Probably safe to strip it out if you want.

Reply:
If you check `man 2 open` on Linux, you'll see that 2 forms of the `open` function exist, one that takes `mode`.  I found that in testing / experimentation with larger binaries, that if `mode` is removed, Manticore gets runtime errors when it hits that form of syscall.  Additionally, if its type is listed just as `int`, `typeguard` complains at times.

The syscall invocation mechanism in Manticore is actually pretty subtle; I wonder now if `mode` should _actually_ be defined via varargs... but I'll leave that investigation for future work.